misk-cron/src/main/kotlin/misk/cron/CronTask.kt

```diff
@@ -16,14 +17,18 @@ internal class CronTask @Inject constructor() : AbstractIdleService() {
   @Inject private lateinit var cronManager: CronManager
   @Inject private lateinit var leaseManager: LeaseManager
   @Inject @ForMiskCron private lateinit var taskQueue: RepeatedTaskQueue
+  @Inject private lateinit var clusterWeight: ClusterWeightProvider
 
   override fun startUp() {
-    val lease = leaseManager.requestLease(CRON_CLUSTER_LEASE_NAME)
     var lastRun = clock.instant()
+    taskQueue.scheduleWithBackoff(INTERVAL) {
+      if (clusterWeight.get() > 0) {
+        Status.OK
+      }
```

Comment:
I think you need to explicitly return from this lambda, or put everything else below this in an `else` block?

Reply:
Agreed, it will need the return or else block.